common/yak/java/java2ssa/visit_stmt_block_expr.go

```diff
@@ -625,7 +625,11 @@ func (y *builder) VisitExpression(raw javaparser.IExpressionContext) ssa.Value {
 		// 处理赋值表达式的等于号
 		if s := ret.Identifier(0); s != nil {
 			recoverRange := y.SetRange(s)
-			variable = y.CreateVariable(s.GetText())
+			name := s.GetText()
+			variable = y.CreateVariable(name)
+			if y.ReadSelfMemberVariable(name) != nil {
+				variable = y.CreateMemberCallVariable(y.ReadValue("this"), y.EmitConstInst(name))
```

Comment:
`y.ReadValue("this")` will create undefine-this in static method. should check `y.PeekValue` first.